i2c/ssd1306_i2c/ssd1306_i2c.c

```diff
@@ -387,13 +387,31 @@ int main() {
         "by the name of",
         "    PICO"
     };
-
+    int max_lines = SSD1306_HEIGHT / 8;  // Each line is 8px tall.
     int y = 0;
-    for (uint i = 0 ;i < count_of(text); i++) {
+    for (uint i = 0; i < count_of(text); i++) {
+        int line_number = i + 1;
         WriteString(buf, 5, y, text[i]);
-        y+=8;
+        y += 8;
+        // On the 32px-tall versions of SSD1306, we can't display
+        // all the text at once. We need to display 4 lines at a time.
+        bool is_end_of_screen = (line_number % max_lines) == 0;
+        if (is_end_of_screen) {
+            render(buf, &frame_area);
+            sleep_ms(3000);
+            memset(buf, 0, SSD1306_BUF_LEN);
+            y = 0;
+        }
+        // If it's the last line...
+        if (line_number == count_of(text)) {
+            // And the last line is not a multiple of max_lines...
+            if (line_number % max_lines != 0) {
```

Comment:
**If** I'm understanding the code correctly, I guess
```c
            // And the last line is not a multiple of max_lines...
            if (line_number % max_lines != 0) {
```
could be replaced by:
```c
            // And there's pending text in buf...
            if (y != 0) {
```
?
And that'd then allow this part of the code to be moved to _after_ the `for` loop? :thinking: (which would remove the `if (line_number == count_of(text))` test on every loop-iteration)

Reply:
Ah yes this can probably be simplified by relying on `y`, let me revisit

Reply:
OK, much simpler now. Please take another look @lurch